phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/ViewEngines.java

```diff
@@ -159,8 +155,14 @@ private void showEngineDetails(EngineDTO engineDTO) {
         showRightView(currentEnginePanel);
     }
 
-    public void setOnSelectCategory(Consumer<EngineCategoryDTO> onSelectCategory) {
-        this.onSelectCategory = onSelectCategory;
+    private void processFilterText(String filterText) {
```

Comment:
I'm not sure what you mean with functionality. This method, `processFilterText`, is passed to the `SearchBox` as a consumer to be called when an entry has been made.
We can't handle the filter text processing the same for every tab, because there are some slight differences.